app/scripts/directives/configEditor.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .directive('configEditor', function(APIService, DataService) {
+    return {
+      restrict: 'E',
+      scope: {
+        mobileClient: '<',
+      },
+      templateUrl: 'views/config-editor.html',
+      controller: function($scope) {
+        $scope.context = {namespace: _.get($scope, "mobileClient.metadata.namespace")};
+        //keep list of active $scope.services upto date
+        DataService.watch(
+          APIService.getPreferredVersion('serviceinstances'), 
+          $scope.context, 
+          function (serviceinstances){
+            $scope.services = [];
+            _.each(serviceinstances.by("metadata.name"), function(service, serviceName){
+              if(_.get(service, "metadata.labels.mobile") !== "enabled") {
+                return;
+              }
+              if(_.indexOf($scope.mobileClient.spec.excludedServices, serviceName) === -1){
+                $scope.services.push(_.get(service, "metadata.labels.serviceName"));
+              }
+            });
+            DataService.list(APIService.getPreferredVersion('configmaps'), $scope.context, updateConfig, {errorNotification: false});
+          }, 
+          { errorNotification: false }
+        );
+        DataService.watch(APIService.getPreferredVersion('configmaps'), $scope.context, updateConfig, {errorNotification: false});
```

Comment:
Is it possible to add a label selector here? All the configs we are interested in will be labelled mobile:enabled

Reply:
I don't see a way of filtering this, other than in the callback